src/lib.rs

```diff
@@ -44,7 +61,11 @@ async fn unzip_file(
     #[cfg(unix)]
     set_unix_permissions(&outpath, directory, &file).await;
 
-    Ok(())
+    FileUnzipResult {
```

Comment:
We can have a different type of error for this instead of having Option for the file.

Reply:
True! I don't know how much work I will be able to do this week, but feel free to change it :)

Reply:
Done!